connector/src/main/java/org/geysermc/connector/network/translators/world/block/entity/SkullBlockEntityTranslator.java

```diff
@@ -68,4 +84,114 @@ public CompoundTag getDefaultBedrockTag(String bedrockId, int x, int y, int z) {
         tagBuilder.byteTag("SkullType", (byte) 0);
         return tagBuilder.buildRootTag();
     }
+
+    public static SerializedSkin getSkin(com.github.steveice10.opennbt.tag.builtin.CompoundTag tag, GeyserSession session) {
+        if (tag.contains("Owner") && !session.getSkullCache().containsKey(tag)) {
+            com.github.steveice10.opennbt.tag.builtin.CompoundTag Owner = tag.get("Owner");
+            com.github.steveice10.opennbt.tag.builtin.CompoundTag Properties = Owner.get("Properties");
+            ListTag Textures = Properties.get("textures");
+            LinkedHashMap tag1 = (LinkedHashMap) Textures.get(0).getValue();
+            StringTag texture = (StringTag) tag1.get("Value");
+            byte[] decoded = Base64.getDecoder().decode(texture.getValue().getBytes());
+            String url = new String(decoded);
+
+            ObjectMapper objectMapper = new ObjectMapper();
+            try {
+                JsonNode jsonNodeRoot = objectMapper.readTree(url);
+                JsonNode jsonNodeUrl = jsonNodeRoot.get("textures").get("SKIN").get("url");
+                CompletableFuture<SkinProvider.Skin> skinCompletableFuture = SkinProvider.requestSkin(UUID.randomUUID(), jsonNodeUrl.asText(), false);
+                SkinProvider.Skin skin = skinCompletableFuture.get();
+                SerializedSkin serializedSkin = SerializedSkin.of(
+                        "Steve", SkinProvider.SkinGeometry.getSkull().getGeometryName(), ImageData.of(skin.getSkinData()), Collections.emptyList(),
+                        ImageData.EMPTY, SkinProvider.SkinGeometry.getSkull().getGeometryData(), "", true, false, false, "", UUID.randomUUID().toString()
+                );
+                return serializedSkin;
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        } else {
+            return null;
+        }
+        return null;
+    }
+
+    public static void spawnPlayer(GeyserSession session, com.github.steveice10.opennbt.tag.builtin.CompoundTag tag, BlockState blockState) {
+        SerializedSkin skin = getSkin(tag, session);
+        float x = (int) tag.get("x").getValue() + .5f;
+        float y = (int) tag.get("y").getValue() - .01f;
+        float z = (int) tag.get("z").getValue() + .5f;
+        float rotation = 0f;
+
+        if (BlockStateValues.getSkullRotation(blockState) == -1) {
+            y += 0.25f;
+            switch (blockState.getId()) {
+                case 6030: //North
+                    rotation = 180f;
+                    z += 0.24;
+                    break;
+                case 6031: //South
+                    rotation = 0;
+                    z -= 0.24f;
+                    break;
+                case 6032: //West
+                    rotation = 90;
+                    x += 0.24f;
+                    break;
+                case 6033: //East

```

Comment:
These block ids need to be stored somewhere, and since these are Java Edition block IDs, they need to be set at startup time since these IDs will most likely change from version to version.